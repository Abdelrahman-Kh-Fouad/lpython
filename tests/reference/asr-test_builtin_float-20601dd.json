{
    "basename": "asr-test_builtin_float-20601dd",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/test_builtin_float.py",
    "infile_hash": "98ef04ee76f3e5b962137f5df3dedf69fbb8c2db89e1c67ffee1bd47",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-test_builtin_float-20601dd.stdout",
    "stdout_hash": "7dc4e34acbd262534e417e6ff55ef250e9f4808d03270bc7f9cc6e52",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}