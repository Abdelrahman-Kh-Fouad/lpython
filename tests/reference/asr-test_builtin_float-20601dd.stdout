(TranslationUnit (SymbolTable 1 {_lpython_main_program: (Subroutine (SymbolTable 4 {}) _lpython_main_program [] [(SubroutineCall 1 test_float () [] ())] Source Public Implementation () .false. .false.), main_program: (Program (SymbolTable 3 {}) main_program [] [(SubroutineCall 1 _lpython_main_program () [] ())]), test_float: (Subroutine (SymbolTable 2 {i: (Variable 2 i Local () () Default (Integer 4 []) Source Public Required .false.)}) test_float [] [(= (Var 2 i) (ConstantInteger 34 (Integer 4 [])) ()) (Assert (Compare (ConstantReal 0.000000 (Real 8 [])) Eq (ConstantReal 0.000000 (Real 8 [])) (Logical 4 []) (ConstantLogical .true. (Logical 4 [])) ()) ()) (Assert (Compare (Cast (ConstantInteger 34 (Integer 4 [])) IntegerToReal (Real 8 []) (ConstantReal 34.000000 (Real 8 []))) Eq (ConstantReal 34.000000 (Real 8 [])) (Logical 4 []) (ConstantLogical .true. (Logical 4 [])) ()) ()) (Assert (Compare (Cast (Var 2 i) IntegerToReal (Real 8 []) ()) Eq (ConstantReal 34.000000 (Real 8 [])) (Logical 4 []) () ()) ()) (= (Var 2 i) (UnaryOp USub (ConstantInteger 4235 (Integer 4 [])) (Integer 4 []) (ConstantInteger -4235 (Integer 4 []))) ()) (Assert (Compare (Cast (UnaryOp USub (ConstantInteger 4235 (Integer 4 [])) (Integer 4 []) (ConstantInteger -4235 (Integer 4 []))) IntegerToReal (Real 8 []) (ConstantReal -4235.000000 (Real 8 []))) Eq (UnaryOp USub (ConstantReal 4235.000000 (Real 8 [])) (Real 8 []) (ConstantReal -4235.000000 (Real 8 []))) (Logical 4 []) (ConstantLogical .true. (Logical 4 [])) ()) ()) (Assert (Compare (Cast (ConstantInteger 34 (Integer 4 [])) IntegerToReal (Real 8 []) (ConstantReal 34.000000 (Real 8 []))) Eq (ConstantReal 34.000000 (Real 8 [])) (Logical 4 []) (ConstantLogical .true. (Logical 4 [])) ()) ()) (Assert (Compare (Cast (UnaryOp USub (ConstantInteger 4235 (Integer 4 [])) (Integer 4 []) (ConstantInteger -4235 (Integer 4 []))) IntegerToReal (Real 8 []) (ConstantReal -4235.000000 (Real 8 []))) Eq (UnaryOp USub (ConstantReal 4235.000000 (Real 8 [])) (Real 8 []) (ConstantReal -4235.000000 (Real 8 []))) (Logical 4 []) (ConstantLogical .true. (Logical 4 [])) ()) ()) (Assert (Compare (Cast (Cast (ConstantLogical .true. (Logical 1 [])) LogicalToInteger (Integer 4 []) (ConstantInteger 1 (Integer 4 []))) IntegerToReal (Real 8 []) (ConstantReal 1.000000 (Real 8 []))) Eq (ConstantReal 1.000000 (Real 8 [])) (Logical 4 []) (ConstantLogical .true. (Logical 4 [])) ()) ()) (Assert (Compare (Cast (Cast (ConstantLogical .false. (Logical 1 [])) LogicalToInteger (Integer 4 []) (ConstantInteger 0 (Integer 4 []))) IntegerToReal (Real 8 []) (ConstantReal 0.000000 (Real 8 []))) Eq (ConstantReal 0.000000 (Real 8 [])) (Logical 4 []) (ConstantLogical .true. (Logical 4 [])) ()) ())] Source Public Implementation () .false. .false.)}) [])
